neuro_data/pipeline_management/NeuroDataPipelineManagement.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+import datajoint as dj
+
+
+from neuro_data.static_images.data_schemas import StaticScanCandidate, StaticScan, ConditionTier, Frame, InputResponse, Eye, Treadmill, StaticMultiDataset, StaticMultiDatasetGroupAssignment
+
+
+class NeuroDataPipelineManagement():
+    def __init__(self):
+        self.pipeline_anatomy = dj.create_virtual_module('pipeline_anatomy', 'pipeline_anatomy')
+        self.pipeline_fuse = dj.create_virtual_module('pipeline_fuse', 'pipeline_fuse')
+        self.pipeline_stimulus = dj.create_virtual_module('pipeline_stimulus', 'pipeline_stimulus')
+
+    def manually_insert_area_for_scan(self, target_scan, area):
+        neuron_unit_keys = (self.pipeline_fuse.ScanSet().Unit() & target_scan).fetch('KEY')
+        for neuron_unit_key in neuron_unit_keys:
+            neuron_unit_key['brain_area'] = area
+            self.pipeline_anatomy.AreaMembership().insert1(neuron_unit_key, allow_direct_insert=True)
+
+    def manually_insert_layer_for_scan(self, target_scan, layer):
+        neuron_unit_keys = (self.pipeline_fuse.ScanSet().Unit() & target_scan).fetch('KEY')
+        for neuron_unit_key in neuron_unit_keys:
+            neuron_unit_key['layer'] = layer
+            self.pipeline_anatomy.LayerMembership().insert1(neuron_unit_key, allow_direct_insert=True)
+
+    # populate functions
+    def process_static_scan(self, target_scan):
+        print('Running preprocessing checks for ', target_scan)
+
+        # Check if the scan has been processed completely
+        if (self.pipeline_fuse.ScanDone() & target_scan).fetch().size == 0:
+            print('[Preprocessing Check]: ' + str(target_scan) + ' Scan has not been processed yet, please looked into pipeline for details')
+            return
+        else:
+            print('[Preprocessing Check]: ScanDone Check Passed')
+
+        # Check if neurons area are labeled
+        if (self.pipeline_anatomy.AreaMembership() & target_scan).fetch().size == 0:
+            print('[Preprocessing Check]: ' + str(target_scan) + " AreaMembership is not populated")
+            user_input = None
+            while user_input not in ['y', 'n']:
+                user_input = input('Should we manually insert if the area are known and all the same? [y/n]')
+                
+                if user_input == 'y':
+                    area = input('Input area to label neurons with:')
```

Comment:
can you add "(pulse exit to leave)" or something so they know about the exit option

Reply:
Yeah, good point. I added that feature but forgot to change the message.